packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts

```diff
@@ -146,6 +150,9 @@ function SpotProvisioningSpecificationPropertyToJson(property?: EmrCreateCluster
   if (!property) {
     return undefined;
   }
+  if (!cdk.Token.isUnresolved(property.timeoutDurationMinutes) && (property.timeoutDurationMinutes < 5 || property.timeoutDurationMinutes > 1440)) {
```

Comment:
Unrelated to your PR, but I can't believe we have a prop called `timeoutDurationMinutes: number`. It should have been typed and named as `timeout: Duration`. Wonder if you're willing to deprecate `timeoutDurationMinutes` in favor of a new prop in a separate PR

Reply:
@kaizencc 

> Wonder if you're willing to deprecate timeoutDurationMinutes in favor of a new prop in a separate PR

OK, I will take it.

Reply:
@kaizencc 

Oh, this param `timeoutDurationMinutes` is required. This means that even if we have a new parameter, we still need to specify this parameter, and making it optional would be a breaking change.

What should we do?

Reply:
@kaizencc 

Oh, this param `timeoutDurationMinutes` is required in the interface. ~~This means that even if we have a new parameter, we still need to specify this parameter, and making it optional would be a breaking change.~~

I was mistaken: making a required parameter optional for an interface that is a props is not a breaking change.

But making the new parameter require is a breaking change.

What should we do? Should it be added as a new optional parameter, whereas before it was required?

Reply:
I'll submit a PR anyway.

Reply:
@kaizencc 

I have submitted the PR.

https://github.com/aws/aws-cdk/pull/28532#issue-2060755266